src/simperium/channel.js

```diff
@@ -102,42 +101,56 @@ internal.removeAndSend = function( id ) {
 	return this.store.get( id ).then( remove );
 };
 
-// We've receive a full object from the network. Update the local instance and
-// notify of the new object version
+/**
+ * Updates the ghost store with the updated ghost data based on the data that
+ * comes from applying the patch to the original.
+ *
+ * @param { string } id - the bucket object key for the object being updated
+ * @param { number } version - the version number the ghost is being updated to
+ * @param { Object } data - the new data for the ghost for this version
+ * @param { Object } original - the original data for the ghost before patch is applied
+ * @param { Object } patch - the patch applied to the original to get the resulting data
+ * @param { Object } [acknowledged ] - the change sent by this client matching the received network change
+ * @returns { Promise<*> } resolves once the ghost has been updated
+ */
 internal.updateObjectVersion = function( id, version, data, original, patch, acknowledged ) {
-	var notify,
-		changes,
-		change,
-		patch,
-		localModifications,
-		remoteModifications,
-		transformed,
-		update;
+	const save = () => this.store.put( id, version, data );
+
+	if ( acknowledged ) {
```

Comment:
I think we should call the subscriber api in all cases because the change can different if the server is merging it over another change.